### ISSUE TITLE
subscription messages should trigger a "message" event, but don't.

### ISSUE THREAD

**Issue by enolan** [floating/ethereum-provider#7]
Here is an example taken from EIP-1193:

```javascript

// Example 4: Log new blocks
ethereum
  .request({
    method: 'eth_subscribe',
    params: ['newHeads'],
  })
  .then((subscriptionId) => {
    ethereum.on('message', (message) => {
      if (message.type === 'eth_subscription') {
        const { data } = message;
        if (data.subscription === subscriptionId) {
          if (typeof data.result === 'string' && data.result) {
            const block = data.result;
            console.log(`New block ${block.number}:`, block);
          } else {
            console.error(`Something went wrong: ${data.result}`);
          }
        }
      }
    });
  })
  .catch((error) => {
    console.error(
      `Error making newHeads subscription: ${error.message}.
       Code: ${error.code}. Data: ${error.data}`
    );
  });
```

This won't log anything. If I instead listen for an `"eth_subscription"` event then I get the right objects.

The part where it checks that the result field is a string is [an error in the example](https://github.com/ethereum/EIPs/issues/2319#issuecomment-734169215) as far as I can tell.

**Comment by floating** [comment]
Thanks @enolan, I will look into this!

**Comment by floating** [comment]
Thanks again @enolan! This has been fixed with https://github.com/floating/ethereum-provider/commit/9c58a731a1aa5d283745597821e0f21a3583d830 and is available in `ethereum-provider@0.1.5` and `eth-provider@0.4.2`

**Comment by mahnazghanbar197** [comment]
> Here is an example taken from EIP-1193:
> 
> ```javascript
> 
> // Example 4: Log new blocks
> ethereum
>   .request({
>     method: 'eth_subscribe',
>     params: ['newHeads'],
>   })
>   .then((subscriptionId) => {
>     ethereum.on('message', (message) => {
>       if (message.type === 'eth_subscription') {
>         const { data } = message;
>         if (data.subscription === subscriptionId) {
>           if (typeof data.result === 'string' && data.result) {
>             const block = data.result;
>             console.log(`New block ${block.number}:`, block);
>           } else {
>             console.error(`Something went wrong: ${data.result}`);
>           }
>         }
>       }
>     });
>   })
>   .catch((error) => {
>     console.error(
>       `Error making newHeads subscription: ${error.message}.
>        Code: ${error.code}. Data: ${error.data}`
>     );
>   });
> ```
> 
> This won't log anything. If I instead listen for an `"eth_subscription"` event then I get the right objects.
> 
> The part where it checks that the result field is a string is [an error in the example](https://github.com/ethereum/EIPs/issues/2319#issuecomment-734169215) as far as I can tell.
